src/pyfmi/fmi_base.pyx

```diff
@@ -0,0 +1,788 @@
+        values. If called without argument then the options class for the
+        default estimation algorithm will be returned.
+
+        Parameters::
+
+            algorithm --
+                The algorithm for which the options class should be fetched.
+                Possible values are: 'SciEstAlg'.
+                Default: 'SciEstAlg'
+
+        Returns::
+
+            Options class for the algorithm specified with default values.
+        """
+        return self._default_options('pyfmi.fmi_algorithm_drivers', algorithm)
+
+    def get_last_result_file(self):
+        """ Returns absolute path to the simulation result file for the last simulation if
+            the fmu has been simulated. Otherwise returns None."""
+        return os.path.abspath(self._result_file) if isinstance(self._result_file, str) else None
+
+    def get_log_filename(self):
+        """ Returns a name of the logfile, if logging is done to a stream it returns
+            a string formatted base on the model identifier.
+        """
+        return '{}_log'.format(self._modelId) if self._log_is_stream else decode(self._fmu_log_name)
+
+    def get_number_of_lines_log(self):
+        """
+            Returns the number of lines in the log file.
+            If logging was done to a stream, this function returns 0.
+        """
+        num_lines = 0
+        if self._fmu_log_name != NULL:
+            with open(self._fmu_log_name,'r') as file:
+                num_lines = sum(1 for line in file)
+        return num_lines
+
+    def print_log(self, start_lines=-1, end_lines=-1):
+        """
+        Prints the log information to the prompt.
+        """
+        cdef int N
+        log = self.get_log(start_lines, end_lines)
+        N = len(log)
+
+        for i in range(N):
+            print(log[i])
+    
+    def _get_module_name(self):
+        return "Model"
+
+    def extract_xml_log(self, file_name=None):
+        """
+        Extract the XML contents of a FMU log and write as a new file.
+        If logging was done to a stream, it needs to support 'seek', otherwise an FMUException is raised
+        when invoking this function.
+
+        Parameters::
+
+            file_name --
+                Name of the file which holds the extracted log, or a stream to write to
+                that supports the function 'write'. Default behaviour is to write to a file.
+                Default: get_log_filename() + xml
+
+        Returns::
+            If extract to a file:
+                file_path -- path to extracted XML file
+            otherwise function returns nothing
+        """
+        # XXX: Needs to be here for now, due to circular dependency, since pyfmi.common.log.parser imports FMUException (from this file)
+        from pyfmi.common.log import extract_xml_log
+
+        if file_name is None:
+            file_name = "{}.{}".format(os.path.splitext(self.get_log_filename())[0], 'xml')
+
+        module_name = self._get_module_name()
```

Comment:
was previously 

`model_name = "Model"`, unless `instance(self, FMUModelCS1)`

Replaced by this to get rid of the circular dependency with fmi1.pyx (`FMUModelCS1` overwrites the function)